[iOS] A couple of inline prediction layout tests are flaky

whsieh opened the pull request:
#### f2815ca5349748843abc9a3c2b3b675b359df489
<pre>
[iOS] A couple of inline prediction layout tests are flaky
<a href="https://bugs.webkit.org/show_bug.cgi?id=267639">https://bugs.webkit.org/show_bug.cgi?id=267639</a>
<a href="https://rdar.apple.com/121118748">rdar://121118748</a>

Reviewed by Megan Gardner.

Attempt to triage a couple of flaky layout tests that exercise inline text predictions. I wasn&apos;t
able to reproduce any test failures when running either of these (running them fully parallel for
several hundred iterations, or when running them in order along with the rest of the test group for
several dozen iterations).

These are purely speculative adjustments to the flaky tests.

* LayoutTests/editing/input/ios/typing-with-inline-predictions-expected.txt:
* LayoutTests/editing/input/ios/typing-with-inline-predictions.html:

Instead of only waiting for `typeCharacter`, wait for the editing container to receive an `keyup`
event as well before trying to send the next set of key events.

Also, temporarily add a 100 ms delay between keystrokes. If this patch fixes the flakiness observed
in our automation, we should try removing this delay and see if these tests start flaking again. If
so, we likely have another source of flakiness when synthesizing key events, which the above change
to wait for the `keyup` event does not address.

Lastly, add a `console.log` here so that if the test begins to time out due to never observing an
`keyup` event, we can easily tell.

* LayoutTests/editing/selection/ios/tap-to-change-selection-after-accepting-inline-prediction.html:

Same as above, but additionally remove the `allowsInlinePredictions=true` test option. This test
doesn&apos;t actually need this option, because it explicitly sets inline predictions; I think it&apos;s
possible that, by enabling inline predictions, the real system predictions (or lack thereof)
override the custom prediction (&quot;I want to c&lt;elebrate&gt;&quot;).

Canonical link: <a href="https://commits.webkit.org/273149@main">https://commits.webkit.org/273149@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3614180ddbf1a9d297f0f7045bf3e6d7b3eb30b2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/34413 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/13227 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/36411 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/37097 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/31126 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/15634 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/10337 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/30125 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/34930 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/11208 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/30688 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/9782 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/9882 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/30741 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/38380 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/31271 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/31057 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/35942 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/9992 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/7868 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/33886 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/11798 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/30364 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7918 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/10544 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/10840 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (commented):
LGTM

megangardner reviewed (approved):
I'd like a radar to remove the delay, but it sounds like that's a near-term plan so I won't insist.

whsieh committed:
[iOS] A couple of inline prediction layout tests are flaky
https://bugs.webkit.org/show_bug.cgi?id=267639
rdar://121118748

Reviewed by Megan Gardner.

Attempt to triage a couple of flaky layout tests that exercise inline text predictions. I wasn't
able to reproduce any test failures when running either of these (running them fully parallel for
several hundred iterations, or when running them in order along with the rest of the test group for
several dozen iterations).

These are purely speculative adjustments to the flaky tests.

* LayoutTests/editing/input/ios/typing-with-inline-predictions-expected.txt:
* LayoutTests/editing/input/ios/typing-with-inline-predictions.html:

Instead of only waiting for `typeCharacter`, wait for the editing container to receive an `keyup`
event as well before trying to send the next set of key events.

Also, temporarily add a 100 ms delay between keystrokes. If this patch fixes the flakiness observed
in our automation, we should try removing this delay and see if these tests start flaking again. If
so, we likely have another source of flakiness when synthesizing key events, which the above change
to wait for the `keyup` event does not address.

Lastly, add a `console.log` here so that if the test begins to time out due to never observing an
`keyup` event, we can easily tell.

* LayoutTests/editing/selection/ios/tap-to-change-selection-after-accepting-inline-prediction.html:

Same as above, but additionally remove the `allowsInlinePredictions=true` test option. This test
doesn't actually need this option, because it explicitly sets inline predictions; I think it's
possible that, by enabling inline predictions, the real system predictions (or lack thereof)
override the custom prediction ("I want to c<elebrate>").

Canonical link: https://commits.webkit.org/273149@main
